Code refactoring

krmahadevan opened the pull request:
* Prefer Streams over conventional iterations.
* Remove unwanted code-blocks.

juherr commented on src/main/java/org/testng/TestRunner.java:
```diff
@@ -689,83 +689,90 @@ public int compareTo(@Nonnull IWorker<ITestNGMethod> other) {
     m_allTestMethods = runMethods.toArray(new ITestNGMethod[0]);
   }
 
+  private Comparator<ITestNGMethod> newComparator(boolean needPrioritySort) {
```
static

krmahadevan replied:
fixed

juherr commented on src/main/java/org/testng/TestRunner.java:
```diff
@@ -689,83 +689,90 @@ public int compareTo(@Nonnull IWorker<ITestNGMethod> other) {
+  }
+
+  // If any of the test methods specify a priority other than the default, we'll need to be able to sort them.
+  private BlockingQueue<Runnable> newQueue(boolean needPrioritySort) {
```
static

krmahadevan replied:
fixed

juherr commented on src/main/java/org/testng/internal/MethodHelper.java:
```diff
@@ -461,16 +454,14 @@ public static void dumpInvokedMethodsInfoToConsole(
-        break;
-      }
-    }
+    methods.stream()
```
Try

```java
return methods.stream()
        .filter(method -> methodName.equals(method.getName()))
        .findFirst()
        .ifPresent(method -> calculateMethodCanonicalName(method)) // calculateMethodCanonicalName can return null
        .orElse(null);
```

krmahadevan replied:
`ifPresent()` and `orElse()` I don't think work together, because `ifPresent()` returns `null`

juherr replied:
Right!

```java
return methods.stream()
        .filter(method -> methodName.equals(method.getName()))
        .findFirst()
        .flatMap(method -> calculateMethodCanonicalName(method)) // calculateMethodCanonicalName can return null
        .orElse(null);
```

krmahadevan replied:
Fixed.

krmahadevan commented:
@juherr - I have addressed your review comments. Please let me know if this can be merged. I would like to start off on fixing #326 by building on top of this changes.

krmahadevan commented:
ping @juherr - Please check now.
